homeassistant/components/media_player/plex.py

```diff
@@ -363,14 +365,22 @@ def _clear_media_details(self):
 
     def refresh(self, device, session):
         """Refresh key device data."""
+        import plexapi.exceptions
+
         # new data refresh
         self._clear_media_details()
 
         if session:  # Not being triggered by Chrome or FireTablet Plex App
             self._session = session
+            if not device and len(session.players) == 1:
+                device = session.players[0]
         if device:
             self._device = device
-            if "127.0.0.1" in self._device.url("/"):
+            try:
+                device_url = self._device.url("/")
+            except plexapi.exceptions.BadRequest:
```

Comment:
It raises BadRequest when you're generating a url ?

Reply:
In this case yes, as the device does not have a root URL associated with it but an empty string. See https://github.com/pkkid/python-plexapi/blob/master/plexapi/client.py#L213